python/ray/tune/trial_runner.py

```diff
@@ -462,14 +462,15 @@ def step(self):
 
         # This will contain the next trial to start
         next_trial = self._get_next_trial()  # blocking
-
-        # Create pending trials
-        num_pending_trials = len(
-            [t for t in self._trials if t.status == Trial.PENDING])
-        while num_pending_trials < self._max_pending_trials:
-            if not self._update_trial_queue(blocking=False):
-                break
-            num_pending_trials += 1
+        # Create pending trials. Skip if next_trial is None (then there
+        # are no new trials to create)
+        if next_trial:
+            num_pending_trials = len(
+                [t for t in self._trials if t.status == Trial.PENDING])
+            while num_pending_trials < self._max_pending_trials:
+                if not self._update_trial_queue(blocking=False):
+                    break
+                num_pending_trials += 1
```

Comment:
Hm, generally I agree. The reason why I introduced this is that at least one test was failing because of this. It was a test that counted the mocked calls of `self._search_alg.next_trial()`. Generally ensuring a next_trial exists will prevent us from calling `self._search_alg.next_trial()` twice in the same `step()`.

I guess there's two possibilities here:

1. Set a flag each `step()` indicating if `update_trial_queue` was called already and take that instead of `if next_trial`
2. Update the trial queue regardless of the next_trial and change the test. This implies that `self._search_alg.next_trial()` might be called twice if it doesn't create a new trial. Is this acceptable? (I don't see a big reason why it shouldn't, except for the test that is)

Reply:
FWIW I'll go with 1) first but am happy to change to 2)